### PR TITLE
Delete user

### PR DESCRIPTION
**Pull request by aLeuleu**
# DELETE /user/delete-account/ 
### take a look a this beautiful database 
Screenshot 2024-02-17 at 01 25 41

### CODE REVIEW

File: user_management/src/user/views/delete_account.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    user.avatar = None
+    user.save()
+
+
+def delete_tournament(user, access_token):
+    request = InternalRequests.delete(f'{settings.TOURNAMENT_URL}tournament/', headers={'Authorization': access_token})
+    if request.status_code != 200:
+        raise Exception(f'Error deleting tournaments : {request.json()}')
+    request = InternalRequests.post(f'{settings.TOURNAMENT_URL}tournament/player/anonymize/',
+                                    headers={'Authorization': access_token})
```
[the remark this answers is not in the document]
Comment (aLeuleu) [reply user_management/src/user/views/delete_account.py:27]: Ethan said the opposite !.. i'll check again with him

File: user_management/src/user_management/JWTManager.py [hunk 2]
```diff
@@ -47,6 +47,12 @@ def authenticate(encoded_jwt: str) -> (bool, int | None, list[str] | None):
             return False, None, ['User does not exist']
         return True, user_id, None
 
+    @classmethod
+    def is_a_deleted_user(cls, user_id):
+        user = User.objects.get(id=user_id)
+        if user.username.startswith('Deleted User'):
+            return True
```
[the remark this answers is not in the document]
Comment (aLeuleu) [reply user_management/src/user_management/JWTManager.py:54]: You can't create a username that start with "Deleted User" ("Username must be alphanumeric").
There are presently no risks, and it works.
I think we need a third person opinion on this, to decide if we change this or not.

File: user_management/src/user/views/delete_account.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (EthanDelage) [reply user_management/src/user/views/delete_account.py:27]: InternalAuthRequest will only be used to anonymize a profile

[the remark this answers is not in the document]
Comment (EthanDelage) [reply user_management/src/user/views/delete_account.py]: 7cf2cabbeb6e7ec180fdb7ef3dcc6f466b29d3db

File: user_management/src/user_management/JWTManager.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tdameros) [reply user_management/src/user_management/JWTManager.py:54]: Oh yes, sorry, I forgot that the username had to be alphanumeric.

[the remark this answers is not in the document]
Comment (EthanDelage) [reply user_management/src/user_management/JWTManager.py:54]: I think putting a Boolean is a more permanent method. 
But for now, if you can't put a space in a username, it's not a problem.

**Review by EthanDelage — APPROVED** [review]

**Review by tdameros — APPROVED** [review]

**Pull request merged**